tests/golden/check-scan-errors/expected.gold

```diff
@@ -0,0 +1,74 @@
+=== Scan errors found ===
+
+  ➥  In file ./check-second-file.md
+     scan error at src:11:1-30:
+
+     ⛀  Annotation "ignore file" must be at the top of markdown or right after comments at the top
+
+
+
+  ➥  In file ./check-scan-errors.md
+     scan error at src:11:1-28:
+
+     ⛀  Annotation "ignore file" must be at the top of markdown or right after comments at the top
+
+
+
+  ➥  In file ./check-scan-errors.md
+     scan error at src:15:1-17:
+
+     ⛀  Expected a PARAGRAPH after "ignore paragraph" annotation, but found  HEADING
+
+
+
+  ➥  In file ./check-scan-errors.md
+     scan error at src:17:1 - 16:0:
+
+     ⛀  Expected a LINK after "ignore link" annotation
+
+
+
+  ➥  In file ./check-scan-errors.md
+     scan error at src:21:1 - 20:0:
+
+     ⛀  Unrecognised option "unrecognised-annotation" perhaps you meant <"ignore link"|"ignore paragraph"|"ignore file">
```

Comment:
Hm, it seems the position info of these 4 errors is not entirely consistent:

* The `Expected a LINK` error points to line 17, where the annotation is declared.
* The `Unrecognised option` error points to line 21, where the annotation is declared.

This behaviour seems correct to me.
However:

* The `Annotation "ignore file" must` error points to line 11, a couple of lines below where the annotation is declared.
* The `Expected a PARAGRAPH` error points to line 15, a couple of lines below where the annotation is declared.


Can you please make these last two errors behave like the first two?

Reply:
Oh just noticed another odd thing:

```
  ➥  In file ./check-scan-errors.md
     scan error at src:17:1 - 16:0:

     ⛀  Expected a LINK after "ignore link" annotation
```

It's saying the error starts in line 17/column 1 and ends in line 16 / column 0? :thinking: I think it should say `src:17:1-31`

Reply:
Oh that's very weird. 
Also, even though both `ingore file` annotations are exactly the same, we get different lengths for these annotations
https://github.com/serokell/xrefcheck/blob/1295eecc205c7bdcbae4a07629887bfcc5484a1a/tests/golden/check-scan-errors/expected.gold#L3-L13
There seems to be a bug in `commonmarkToNode` from the `cmark-gfm` package.

Reply:
It looks like this bug only affects the positions of one line `HTML_BLOCK`s.

Reply:
> Also, even though both ignore file annotations are exactly the same, we get different lengths for these annotations

My bad. This actually is not related to ` commonmarkToNode` bug. This is because we report position of the next node instead of the position of the annotation.